### ISSUE TITLE
SetTitleVisible ignores completion flag from UIView function

### ISSUE THREAD

**Issue by miradesne** [Skyscanner/SkyFloatingLabelTextField#110]
In the code setTitleVisible(animated:):
```
if animated {
            let animationOptions:UIViewAnimationOptions = .curveEaseOut;
            let duration = self.isTitleVisible() ? titleFadeInDuration : titleFadeOutDuration

            UIView.animate(withDuration: duration, delay: 0, options: animationOptions, animations: { () -> Void in
                updateBlock()
                }, completion: { _ in
                    completion?()
                })
        } else {
            updateBlock()
            completion?()
        }
```

The function just ignored the flag from:

```
class func animate(withDuration duration: TimeInterval, 
        animations: @escaping () -> Void, 
        completion: ((Bool *completed*) -> Void)? = nil)
```

and calls the completion call regardless if the animation is cancelled. It caused a bug where we want to only set some thing if the animation is completed (instead of cancelled).

**Comment by k0nserv** [comment]
That seems wrong yes. Is this behaviour documented anywhere ie only calling the completion block when the animation does complete?

**Comment by k0nserv** [comment]
I think we should still be calling the completion, but passing along the `completed` value in the event of a canceled animation. @bogren thoughts?

**Comment by miradesne** [comment]
@k0nserv Yes it should still call the completion closure, but should pass the flag over so that we know how to update UI when we cancel the animation... Thanks!

**Comment by k0nserv** [comment]
Cool, I have an in progress implementation of that. I'll finish it up tonight and release a new version.

**Comment by k0nserv** [comment]
This has been released as 3.0.0
